### PR TITLE
[WIP] - Add a k8s helper method for generating viewer crd for tensorboard.

### PR DESCRIPTION
**Pull request by jingzhang36**
Haven't tested yet. Post it to collect some initial comments.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/1141)

### CODE REVIEW

File: frontend/server/k8s-helper.ts [hunk 1]
```diff
@@ -127,3 +127,36 @@ export function getPodLogs(podName: string): Promise<string> {
       (error: any) => { throw new Error(JSON.stringify(error.body)); }
     );
 }
+
+/**
+* Creates a CRD (Customer Resource Definition) file. This CRD file can be used
+* by CRD controller to create new Tensorboard Pod.
+*/
+export async function newTensorboardCRD(logdir: string): Promise<void> {
+  const crd = {
```
Comment (IronPan) [inline frontend/server/k8s-helper.ts]: is it possible to use k8s js client?
https://github.com/kubernetes-client/javascript

hard coding template is considered not a good practice and we should avoid if possible :)

**Comment by yebrahim** [comment]
Can you explain more what this change does? What does writing a CRD instance to a file on the web server achieve?

File: frontend/server/k8s-helper.ts (hunk 1 above)
Comment (jingzhang36) [reply frontend/server/k8s-helper.ts]: Sure. Let me figure out how to use k8s js client and update my change. If you can help me with some similar examples, that will be really helpful to me :)

**Comment by jingzhang36** [comment]
> Can you explain more what this change does? What does writing a CRD instance to a file on the web server achieve?

The context is we are removing /data endpoint https://github.com/kubeflow/pipelines/issues/1100#event-2270474368. The plan is to let Tensorboard CRD controller handle Tensorboard pods. 

My understanding of the current structure is, from kubeflow UI, Tensorboard pods are created via /data to view output. Later, we want those Tensorboard pods created via Tensorboard CRD controller. And thus when viewing the output in kubeflow UI, we don't create tensorboard pods but generate a crd files; and Tensorboard CRD controller can use the crd file to create/manage Tensorboard pods.

Since I am not very familiar with kubeflow pipeline yet, I start with the small step of generating crd file. Later I'll make change to let UI to call this method to actually generate crd file. 

Moreover, I feel the generated crd files are to be stored in local storage, and thus used logdir. However, I am not sure whether this is the right place, and that's why I ask for some initial comments to clear aways my misunderstandings.

**Comment by jingzhang36** [comment]
Will have a new PR for this task. So close this one.

**Pull request closed without merging**